tests/test_main.py

```diff
@@ -190,3 +191,40 @@ def test_write_results_from_dummy_data_file(dummy_data_fname, results_fname, tmp
     results_file = tmp_path / results_fname
     main.write_results(dummy_data_file, results_file)
     assert results_file.read_text(encoding="utf-8") == "id\n1\n2\n"
+
+
+def test_generate_column_headers(tmp_path):
```

Comment:
I'm in two minds about test-driven development. Whilst it's good in principle to write a failing test, in practice I think we may as well have written the implementation and a passing test. That said, if we _are_ going to write a failing test, then we should mark the test as expected to fail, so we don't inadvertently mask other failures. (To be clear, there are no other failures here. I'm thinking in more general terms.)

```py
@pytest.mark.xfail
def test_generate_column_headers(tmp_path):
    ...
```

As I've mentioned in another comment, I don't think `generate_column_headers` has the right name or signature.